### PR TITLE
feat: debug state for the heatmap chart 

### PR DESCRIPTION
**Pull request by darnautov**
## Summary

Resolves https://github.com/elastic/elastic-charts/issues/959 

- Adds debug state support for the heatmap/swimlane chart 
- Extends `DebugState` interface with heatmap related data under the `heatmap` property 
- Adds debug action and cell selection knobs to the storybook

### CODE REVIEW

File: src/chart_types/heatmap/state/selectors/get_debug_state.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+interface DebugStateAxes {
+  x: DebugStateAxis;
+  y: DebugStateAxis;
+}
+
+export type HeatmapDebugState = Pick<DebugState, 'legend'> & {
+  cells: CellDebug[];
+  selectedArea: { x: number; y: number; width: number; height: number } | null;
+  axes: DebugStateAxes;
+};
```
Comment (nickofthyme) [inline src/chart_types/heatmap/state/selectors/get_debug_state.ts:47]: We had talked about using a single `DebugState` type for all chart types to simplify the API. In that case, we would add these additional keys on the `DebugState` as optional.

@markov00 any thoughts?

Comment (markov00) [reply src/chart_types/heatmap/state/selectors/get_debug_state.ts:47]: I'm also for simplifying the API here if possible

Comment (darnautov) [reply src/chart_types/heatmap/state/selectors/get_debug_state.ts]: hey @nickofthyme @markov00 , I've updated the `DebugState` interface, please take a look at the recent changes

File: src/state/types.ts [hunk 2]
```diff
@@ -78,6 +79,13 @@ export type DebugStateBar = DebugStateBase & {
   labels: any[];
 };
 
+type CellDebug = Pick<Cell, 'value' | 'formatted' | 'x' | 'y'> & { fill: string };
+
+type HeatmapDebugState = {
+  cells: CellDebug[];
+  selectedArea: { x: number; y: number; width: number; height: number } | null;
+};
```
Comment (nickofthyme) [inline src/state/types.ts:87]: 👍🏼

**Review by nickofthyme — APPROVED** [review]
This is exactly what I was thinking. Thanks for the changes.

**Review by markov00 — APPROVED** [review]
Looks good to merge when Jenkins is green

**Pull request merged**